app/services/cocina/from_fedora/descriptive/language.rb

```diff
@@ -37,28 +45,89 @@ def build
 
         attr_reader :ng_xml
 
+        def lang_term_attributes_for(lang)
+          {
+            code: language_code_for(lang),
+            value: language_text_for(lang),
+            uri: language_value_uri_for(lang),
+            appliesTo: language_applies_to(lang),
+            displayLabel: language_display_label(lang),
+            source: language_source_for(lang)
+          }.reject { |_k, v| v.blank? }
+        end
+
+        def script_term_attributes_for(script_term_nodes)
+          return if script_term_nodes.blank?
+
+          code, value, authority = nil
+          script_term_nodes.each do |script_term_node|
+            code ||= script_term_node.xpath('text()').to_s if script_term_node.xpath('@type').to_s == 'code'
```

Comment:
I find that using the methods on the node is easier to understand. What do you think of this?
```suggestion
            code ||= script_term_node.content if script_term_node['type'] == 'code'
```

Reply:
thanks!  this is nicer!